ppr-ui/src/utils/registration-helper.ts

```diff
@@ -22,6 +24,147 @@ import {
+  // Conditionally set draft vehicle collateral
+  if (collateral.vehicleCollateral && collateral.vehicleCollateral.length > 0) {
+    for (let i = 0; i < collateral.vehicleCollateral.length; i++) {
+      if (collateral.vehicleCollateral[i].action) {
+        if (collateral.vehicleCollateral[i].action === ActionTypes.ADDED ||
+            collateral.vehicleCollateral[i].action === ActionTypes.EDITED) {
+          statement.addVehicleCollateral.push(JSON.parse(JSON.stringify(collateral.vehicleCollateral[i])))
+        }
+        if (collateral.vehicleCollateral[i].action === ActionTypes.REMOVED ||
+            collateral.vehicleCollateral[i].action === ActionTypes.EDITED) {
+          statement.deleteVehicleCollateral.push(JSON.parse(JSON.stringify(collateral.vehicleCollateral[i])))
+        }
+      }
+    }
+  }
+  // Conditionally set draft general collateral
+  if (collateral.generalCollateral && collateral.generalCollateral.length > 0) {
+    for (let i = 0; i < collateral.generalCollateral.length; i++) {
+      if (collateral.generalCollateral[i].descriptionAdd &&
+          collateral.generalCollateral[i].descriptionAdd.trim().length > 0) {
+        const gc:GeneralCollateralIF = {
+          description: collateral.generalCollateral[i].descriptionAdd
+        }
+        statement.addGeneralCollateral.push(gc)
+      }
+      if (collateral.generalCollateral[i].descriptionDelete &&
+          collateral.generalCollateral[i].descriptionDelete.trim().length > 0) {
+        const gc:GeneralCollateralIF = {
+          collateralId: collateral.generalCollateral[i].collateralId,
+          description: collateral.generalCollateral[i].descriptionDelete
+        }
+        statement.deleteGeneralCollateral.push(gc)
+      }
+    }
+  }
+  return statement
+}
+
+/** Setup the draft data for the current amendment. All data to be saved is in the store state model. */
+export function setupAmendmentStatementDraft (stateModel:StateModelIF): DraftIF {
+  const draft:DraftIF = stateModel.registration.draft
+  draft.type = DraftTypes.AMENDMENT_STATEMENT
+  draft.amendmentStatement = setupAmendmentStatement(stateModel)
+  return draft
+}
+
+/** Save or update a draft of the current amendment. */
+export async function saveAmendmentStatementDraft (stateModel:StateModelIF): Promise<DraftIF> {
+  const draft:DraftIF = setupAmendmentStatementDraft(stateModel)
+  let draftResponse:DraftIF = null
+  let apiCall:String = ''
+  if (draft.amendmentStatement.documentId !== undefined && draft.amendmentStatement.documentId !== '') {
+    apiCall = 'update'
+    draftResponse = await updateDraft(draft)
+  } else {
+    apiCall = 'create'
+    draftResponse = await createDraft(draft)
+  }
+  if (draftResponse !== undefined && draftResponse.error === undefined) {
+    console.log('saveAmendmentStatementDraft ' + apiCall + ' draft successful for documentId ' +
+                draftResponse.amendmentStatement.documentId)
+  } else if (draftResponse !== undefined) {
```

Comment:
nit: I find this a little confusing, and would be tempted to check draftResponse.error instead... and is there a case where draftResponse would be blank? exception?

Reply:
if you change this to ```if (!draftResponse) ...``` it will catch it if it is undefined, null, empty etc. Would that be better here and just above?

Reply:
Updated in commit, allowing for case where draftResponse is null.